### PR TITLE
Add float16 support to batch norm operator

### LINKED ISSUE #9175: float16 support for batch norm operator

### PR DESCRIPTION
**Pull request by kexinzhao**
fix #9175 

Only added and verified float16 kernel for the inference mode of cudnn batch norm kernel,
which is needed to run vgg/resnet inference.

OpTest.np_dtype_to_fluid_dtype is to change the dtype of a numpy array from float16 to uint16 so that it can correctly bind with paddle float16 in tensor_py.h.

### CODE REVIEW

File: paddle/fluid/operators/batch_norm_op.cu.cc [hunk 1]
```diff
@@ -118,15 +122,16 @@ class BatchNormKernel<platform::CUDADeviceContext, T>
 
     // alloc memory
     y->mutable_data<T>(ctx.GetPlace());
```
Comment (jacquesqiao) [inline paddle/fluid/operators/batch_norm_op.cu.cc]: why here does not use BatchNormParamType?

Comment (kexinzhao) [reply paddle/fluid/operators/batch_norm_op.cu.cc]: Per the cudnn API, when we are using fp16 mode to run Batch norm cudnn, the data type of input `x` and output `y` will be fp16, but the other input parameters including `mean` and `variance` will still be float32 type.

So here `T == float16`, and `BatchNormParamType<T> == float` as defined in `cudnn_helper.h`
